src/System.Private.CoreLib/shared/System/IO/Path.Windows.cs

```diff
@@ -252,24 +252,21 @@ internal static ReadOnlySpan<char> GetVolumeName(ReadOnlySpan<char> path)
             {
                 // Cut from "\\?\UNC\Server\Share" to "Server\Share"
                 // Cut from  "\\Server\Share" to "Server\Share"
-                return TrimEndingDirectorySeparator(root.Slice(offset));
+                return TrimEndingDirectorySeparatorHelper(root.Slice(offset));
             }
             else if (PathInternal.IsDevice(path))
             {
-                return TrimEndingDirectorySeparator(root.Slice(4)); // Cut from "\\?\C:\" to "C:"
+                return TrimEndingDirectorySeparatorHelper(root.Slice(4)); // Cut from "\\?\C:\" to "C:"
             }
 
-            return TrimEndingDirectorySeparator(root); // e.g. "C:"
-        }
+            return TrimEndingDirectorySeparatorHelper(root); // e.g. "C:"
 
-        /// <summary>
-        /// Trims the ending directory separator if present.
-        /// </summary>
-        /// <param name="path"></param>
-        internal static ReadOnlySpan<char> TrimEndingDirectorySeparator(ReadOnlySpan<char> path) =>
-            PathInternal.EndsInDirectorySeparator(path) ?
-                path.Slice(0, path.Length - 1) :
-                path;
+            // Trim ending directory separator
+            ReadOnlySpan<char> TrimEndingDirectorySeparatorHelper(ReadOnlySpan<char> pathToTrim) =>
```

Comment:
`TrimEndingDirectorySeparatorHelper` is both confusing name (it is not immediately obvious how it is different from `TrimEndingDirectorySeparator`; and unnecessary overhead (extra function call or duplicated code if it gets inlined).

```csharp
int startOffset = GetUncRootLength(path);
if (startOffset >= 0)
{
    // Cut from "\\?\UNC\Server\Share" to "Server\Share"
    // Cut from  "\\Server\Share" to "Server\Share"
}
else if (PathInternal.IsDevice(path))
{
    startOffset = 4; // Cut from "\\?\C:\" to "C:"
}
else
{
    startOffset = 0; // e.g. "C:"
}

// Trim the directory separator unconditionally (different from TrimEndingDirectorySeparator)
ReadOnly<char> pathToTrim = root.Slice(startOffset);
return PathInternal.EndsInDirectorySeparator(pathToTrim) ? pathToTrim.Slice(0, pathToTrim.Length - 1) : pathToTrim;
```

Reply:
Your version looks fine to me too.

Reply:
pushed!